src/editor/adapters/change-signature-webview/createChangeSignatureWebviewTemplate.test.ts

```diff
@@ -42,48 +74,192 @@ describe("Change signature Webview Content", () => {
       });
     });
 
-    it("Should be able to move paramB as a first parameter", async () => {
-      const upButtons = document.querySelectorAll<HTMLElement>(".up");
-      upButtons[1].click();
+    it("Should be able to move paramB as a first parameter", () => {
+      const paramsTr = document.querySelectorAll<HTMLTableRowElement>(".param");
+      paramsTr[1].click();
+      const upButton = document.getElementById("up") as HTMLSpanElement;
+      upButton.click();
       document.getElementById("confirm")?.click();
 
       expect(postMessage).toHaveBeenCalledWith({
         values: [
-          { label: "paramA", startAt: 0, endAt: 1 },
-          { label: "paramB", startAt: 1, endAt: 0 }
+          { label: "paramB", startAt: 1, endAt: 0 },
+          { label: "paramA", startAt: 0, endAt: 1 }
         ]
       });
     });
 
-    it("Should be able to move paramA as last parameter", async () => {
-      const downButtons = document.querySelectorAll<HTMLElement>(".down");
-      downButtons[0].click();
+    it("Should be able to move paramA as last parameter", () => {
+      const paramsTr = document.querySelectorAll<HTMLTableRowElement>(".param");
+      paramsTr[0].click();
+      const downButton = document.getElementById("down") as HTMLSpanElement;
+      downButton.click();
       document.getElementById("confirm")?.click();
 
       expect(postMessage).toHaveBeenCalledWith({
         values: [
-          { label: "paramA", startAt: 0, endAt: 1 },
-          { label: "paramB", startAt: 1, endAt: 0 }
+          { label: "paramB", startAt: 1, endAt: 0 },
+          { label: "paramA", startAt: 0, endAt: 1 }
         ]
       });
     });
 
-    it("Should not change order if I move the last parameter down", async () => {
-      const downButtons = document.querySelectorAll<HTMLElement>(".down");
-      downButtons[1].click();
+    it("Should down button be disabled when I select the last parameter", () => {
+      const paramsTr = document.querySelectorAll<HTMLTableRowElement>(".param");
+      paramsTr[1].click();
+      const downButton = document.getElementById("down") as HTMLSpanElement;
+
+      expect(downButton.classList.contains("disabled")).toBeTruthy();
+    });
+
+    it("Should up button be disabled when I select the first parameter", () => {
+      const paramsTr = document.querySelectorAll<HTMLTableRowElement>(".param");
+      paramsTr[0].click();
+      const upButton = document.getElementById("up") as HTMLSpanElement;
+
+      expect(upButton.classList.contains("disabled")).toBeTruthy();
+    });
+
+    it("Should up button be disabled when I move paramB as a first parameter", () => {
+      const paramsTr = document.querySelectorAll<HTMLTableRowElement>(".param");
+      paramsTr[1].click();
+      const up = document.getElementById("up") as HTMLSpanElement;
```

Comment:
🍰 At this point I feel it would help to extract some helper functions to make the tests easier to read. Something like:

```ts
clickOnParamWithIndex(1);
clickUp();
```

See what I mean?

Reply:
Yes, I introduce both functions at https://github.com/nicoespeon/abracadabra/pull/775/commits/590cac99ea87fb45512ce50d45cd9027c646f349 and https://github.com/nicoespeon/abracadabra/pull/775/commits/ee54e6a4c83e64d76b150340a2398cdeb25bc3bd